### ISSUE TITLE
[Redesign] [Feature Request] Automatic sync in background.

### ISSUE THREAD

**Issue by brainiac19** [jmshrv/finamp#864]
It would be great to have auto-sync as an option. Basically it lets users choose what they'd like to sync in the background while the app's running. This could contain:

- Pick a list of playlists to sync all songs
- Pick a list of artists to sync all songs
- Sync playlists metadata (already implemented but manual)
- Cache all images (already implemented but manual)

**Comment by Chaphasilor** [comment]
Finamp should already synchronize all of this on startup. Would you like this to also happen periodically, or even every time the app is opened? 🤔

I'm also not sure about your mention of "manual". All of these should sync automatically on startup.

**Comment by brainiac19** [comment]
I now realize my description was rather unclear. Allow me to try again.

> Finamp should already synchronize all of this on startup. Would you like this to also happen periodically, or even every time the app is opened? 🤔

Periodically yes, because I gave finamp no battery restriction so it rarely fully exit, and that I also listen and edit playlists from other clients like on Windows. God I wish there's something from the server side that can tell finamp whether stuff have changed since the last sync, but I guess not.

> I'm also not sure about your mention of "manual". All of these should sync automatically on startup.

I was referring to downloading the actual music content as opus, it's my understanding that it only happens when manually clicking download. I was gonna suggest caching stuff while playing the way spotify does, but it's my very uneducated guess that it's gonna be a lot harder to implement than periodic music sync.

**Comment by brainiac19** [comment]
Or I suggest another alternative. 

1. Add option to download selected playlists or artists' music on startup.
2. Add option to treat switching app to foreground as a sync condition.

**Comment by Chaphasilor** [comment]
Thanks for tue clarification!  
My point stands, the actual music files of e.g. tracks newly added to playlists should be downloaded automatically on startup, as long as you at one point downloaded that playlist (either specifically, or by downloading all playlists in the settings). If that isn't working for you, that's a bug that we need to fix. So please try that, and if it's not working, share your download screen contents, download settings, and logs to help diagnose the issue :)


Peridically synching should be doable. Maybe in the future we can also notify the client if changes were made on the server...

For now, depending on your usage, you could re-enable battery restrictions and enable "Enter low priority state on pause" in Audio Service settings, to force Finamp to be killed whenever you pause. If you re-open Finamp it will automatically restore the queue, so you can pick up where you left off, while triggering a sync each time.

**Comment by brainiac19** [comment]
I've had problems with my stupid phone killing spotify so I removed battery restriction for finamp as I switched to it, but I'll try again with your suggested settings. 

Also now I think to achieve the kind of real-time sync I said, relying on polling is just as not ideal as relying on app being restarted. So if one day finamp decides to implement real-time sync, maybe look into what can be done in coordination with the server side. 

Thx for the help!
